doc/guides/zephyr_cmake_package.rst

```diff
@@ -0,0 +1,231 @@
+.. _cmake_pkg:
+
+Zephyr CMake Package
+####################
+
+Zephyr can be exported as a CMake config package.
+
+A CMake config package allows a developed to integrate Zephyr into their own application by using
+``find_package(Zephyr)``.
+
+Exporting Zephyr as a CMake package is identical to create a reference to the current Zephyr
+installation inside the CMake user package registry.
+
+
+.. tabs::
+
+   .. group-tab:: Ubuntu
+
+      In Linux, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: macOS
+
+      In macOS, the CMake user package registry is found in:
+
+      ``~/.cmake/package/Zephyr``
+
+   .. group-tab:: Windows
+
+      In Windows, the CMake user package registry is found in:
+
+      ``HKEY_CURRENT_USER\Software\Kitware\CMake\Packages\Zephyr``
+
+
+The Zephyr package allows CMake to automatically find a Zephyr base.
+At minimum, a single Zephyr installation must be exported, but it is allowed to export multiple.
+Exporting multiple Zephyr installations may be useful when developing applications out-of-tree.
+
+
+Zephyr CMake package export (west)
+**********************************
+
+When installing Zephyr using :ref:`west <get_the_code>` then it is recommended to export Zephyr
+using ``west zephyr-export``.
+
+
+Zephyr CMake package export (without west)
+******************************************
+
+Zephyr can be exported using CMake directly.
+To do so, run the following command:
+
+``cmake -S <PATH-TO-ZEPHYR>/share/zephyr-package/cmake -B <PATH-TO-ZEPHYR>/share/zephyr-package/cmake``
+
+
+this will export the current Zephyr to CMake user package registry.
+
+.. _zephyr_cmake_package_zephyr_base:
+
+Zephyr Base Environment Setting
+*******************************
+
+The Zephyr CMake package search functionality allows for explicitly specifying a Zephyr base using
+an environment variable.
+
+In order to support the use of ``ZEPHYR_BASE`` environment setting, the following ``find_package()``
+syntax much be used:
+
+  .. code-block:: cmake
```

Comment:
```suggestion
.. code-block:: cmake
```

The code-block is an element at the same indentation level as a paragraph. The code inside the code-block is what gets indented.

Reply:
fixed